lib/Parse/ParseDecl.cpp

```diff
@@ -1021,6 +1021,13 @@ bool Parser::parseDifferentiableAttributeArguments(
                         { label });
     result.Name = parseDeclNameRef(result.Loc, funcDiag,
         DeclNameFlag::AllowZeroArgCompoundNames | DeclNameFlag::AllowOperators);
+    // Emit warning for deprecated `jvp:` and `vjp:` arguments.
+    // TODO(TF-1001): Remove deprecated `jvp:` and `vjp:` arguments.
+    if (result.Loc.isValid()) {
```

Comment:
Note: `result.Loc` may not be valid if `parseDeclNameRef` produces an error.
This can happen when `jvp:vjp:` specifies a non-`func` declaration, e.g. an `init` or `subscript`.